File: widget/src/main/java/ru/kest/trainswidget/data/DataService.java
package ru.kest.trainswidget.data;

import android.content.Context;

/**
 * Created by KKharitonov on 14.02.2016.
 */
public class DataService {

    private static DataProvider dataProvider = null;


    public static DataProvider getDataProvider(Context context) {
        if (dataProvider == null) {
            dataProvider = new DataProviderImpl(context);
        }
        return dataProvider;
    }
}

File: widget/src/main/java/ru/kest/trainswidget/data/DataService.kt
package ru.kest.trainswidget.data

import android.content.Context

/**
 * Singleton factory for DataProvider
 *
 * Created by KKharitonov on 14.02.2016.
 */
class DataService(context: Context) {

    val dataProvider: DataProvider by lazy { DataProviderImpl(context) }

}
